src/EnergyPlus/Photovoltaics.cc

```diff
@@ -856,55 +856,53 @@ namespace Photovoltaics {
 		//       AUTHOR         B. Griffith
 		//       DATE WRITTEN   Jan. 2004
 		//       MODIFIED       B. Griffith, Aug. 2008
-		//       RE-ENGINEERED  na
 
 		// PURPOSE OF THIS SUBROUTINE:
 		// collect statements that assign to variables tied to output variables
 
-		// METHODOLOGY EMPLOYED:
-		// <description>
-
-		// REFERENCES:
-		// na
-
-		// USE STATEMENTS:
-		// na
 		// Using/Aliasing
 		using DataHeatBalance::Zone;
+		using DataGlobals::NumOfZones;
 		using DataSurfaces::Surface;
 		using DataHeatBalFanSys::QPVSysSource;
 		using TranspiredCollector::SetUTSCQdotSource;
-
-		// Locals
-		// SUBROUTINE ARGUMENT DEFINITIONS:
-
-		// SUBROUTINE PARAMETER DEFINITIONS:
-		// na
-
-		// INTERFACE BLOCK SPECIFICATIONS:
-		// na
-
-		// DERIVED TYPE DEFINITIONS:
-		// na
+		using InputProcessor::FindItemInList;
 
 		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
 		int thisZone; // working index for zones
+		int thisSurface; // index for PV surface
+		Array1D< bool > PVarrayZoneInitialized; // avoid doing the zone check too often
+		bool FirstTimeThru( true );
 
 		PVarray( PVnum ).Report.DCEnergy = PVarray( PVnum ).Report.DCPower * ( TimeStepSys * SecInHour );
 
+		thisSurface = PVarray( PVnum ).SurfacePtr;
+
+		if ( FirstTimeThru ) {
+			PVarrayZoneInitialized.dimension( NumPVs, false );
+			FirstTimeThru = false;
+		}
+
+		if ( ! PVarrayZoneInitialized( PVnum ) ) {
+			if ( Surface( thisSurface ).Zone == 0 ) { // might need to get the zone number from the name
+				Surface( thisSurface ).Zone = FindItemInList( Surface( thisSurface ).ZoneName, Zone, NumOfZones );
```

Comment:
Why isn't this done in GetInput?

Reply:
@rraustad Good point.  I will change this so that it gets the Zone index directly during GetInput and store it in the local PV data structure.  That should then address what you and Edwin pointed out and still fix the issue.  Thanks!